### PR TITLE
fix: format table name to include backticks

### LINKED ISSUE #47: Cannot open databases with all-numeric names
Using build 0.1.8 in Debian Bookworm under WLS2

I noticed this connected to two similarly configured MySQL servers.
I do not know if it is a MySQL specific problem or not.

I can see all the databases in the table panel.
I can open ones with letters in their names, even if they start with a number.
All-numeric databases do not open. I can select them but nothing happens when I press enter.

I hope this is helpful, thanks!

### PR DESCRIPTION
**Pull request by jorgerojas26**
Fixes #47

### CODE REVIEW

File: drivers/mysql.go [hunk 1]
```diff
@@ -313,6 +320,7 @@ func (db *MySQL) ExecuteQuery(query string) (results [][]string, err error) {
 
 // TODO: Rewrites this logic to use the primary key instead of the id
 func (db *MySQL) UpdateRecord(table, column, value, primaryKeyColumnName, primaryKeyValue string) error {
+	table = db.formatTableName(table)
 	query := fmt.Sprintf("UPDATE %s SET %s = \"%s\" WHERE %s = \"%s\"", table, column, value, primaryKeyColumnName, primaryKeyValue)
```
Comment (ccoVeille) [inline drivers/mysql.go:324]:
```suggestion
 	query := fmt.Sprintf(`UPDATE %s SET %s = "%s" WHERE %s = "%s"`, table, column, value, primaryKeyColumnName, primaryKeyValue)
```

Comment (ccoVeille) [inline drivers/mysql.go:324]: Also, I'm very worried about the  possible SQL injection

You should consider using `?` for value, then pass `value ` to Exec as an arg

File: drivers/mysql.go [hunk 2]
```diff
@@ -321,6 +329,7 @@ func (db *MySQL) UpdateRecord(table, column, value, primaryKeyColumnName, primar
 
 // TODO: Rewrites this logic to use the primary key instead of the id
 func (db *MySQL) DeleteRecord(table, primaryKeyColumnName, primaryKeyValue string) error {
+	table = db.formatTableName(table)
 	query := fmt.Sprintf("DELETE FROM %s WHERE %s = \"%s\"", table, primaryKeyColumnName, primaryKeyValue)
```
Comment (ccoVeille) [inline drivers/mysql.go:333]: Same concerns here

File: drivers/mysql.go [hunk 3]
```diff
@@ -385,7 +394,7 @@ func (db *MySQL) ExecutePendingChanges(changes []models.DbDmlChange, inserts []m
 
 		statementType = "DELETE FROM"
 
-		query = fmt.Sprintf("%s %s WHERE %s = \"%s\"", statementType, delete.Table, delete.PrimaryKeyColumnName, delete.PrimaryKeyValue)
+		query = fmt.Sprintf("%s %s WHERE %s = \"%s\"", statementType, db.formatTableName(delete.Table), delete.PrimaryKeyColumnName, delete.PrimaryKeyValue)
```
Comment (ccoVeille) [inline drivers/mysql.go:397]: Same

File: drivers/mysql.go [hunk 4]
```diff
@@ -405,7 +414,7 @@ func (db *MySQL) ExecutePendingChanges(changes []models.DbDmlChange, inserts []m
 			}
 		}
 
-		query := fmt.Sprintf("INSERT INTO %s (%s) VALUES (%s)", insert.Table, strings.Join(insert.Columns, ", "), strings.Join(values, ", "))
+		query := fmt.Sprintf("INSERT INTO %s (%s) VALUES (%s)", db.formatTableName(insert.Table), strings.Join(insert.Columns, ", "), strings.Join(values, ", "))
```
Comment (ccoVeille) [inline drivers/mysql.go:417]: Here too the values should be passed as ?,?,?...

strings.Repeat could help

And `values` should be passed to Exec

File: drivers/mysql.go [hunk 5]
```diff
@@ -442,3 +449,18 @@ func (db *MySQL) SetProvider(provider string) {
 func (db *MySQL) GetProvider() string {
 	return db.Provider
 }
+
+func (db *MySQL) formatTableName(tableName string) string {
+	splittedTableName := strings.Split(tableName, ".")
+
+	if len(splittedTableName) == 1 {
+		return tableName
```
Comment (ccoVeille) [inline drivers/mysql.go:457]: You could return this


```suggestion
		return fmt.Sprintf("`%s`", table)
```

**Review by ccoVeille — COMMENTED** [review]
I'm sorry because most of my feedbacks are  not related to the current PR.

But I think I had to report them

**Comment by jorgerojas26** [comment]
Thanks for reviewing the PR. May i ask why SQL Injection is a concern to you in this app when it is just a SQL client?.

I totally understand that i should use prepared statements, i plan to make a huge refactor on all of the queries, but i want to solve the problem with a quick solution first. Does it makes sense to you?

**Comment by jorgerojas26** [comment]
The table name formatting is really dirty, but it is a quick solution. That should be refactored too.

**Comment by ccoVeille** [comment]
> Thanks for reviewing the PR. May i ask why SQL Injection is a concern to you in this app when it is just a SQL client?.

I tend to always plan the worst. Often, I'm not disappointed and something close happens 😅

> I totally understand that i should use prepared statements, i plan to make a huge refactor on all of the queries, but i want to solve the problem with a quick solution first. Does it makes sense to you?

Not necessarily prepared statement, but using Exec the right way should be always preferred.

**Comment by ccoVeille** [comment]
Also please consider a table could have a varchar primary key, and that this field may contain a quote

Something like `john "babayaga" wick`

It would be a bad design to rely on a varchar PK for this, but your app will be unable to work with this row

**Comment by ccoVeille** [comment]
> May i ask why SQL Injection is a concern to you in this app when it is just a SQL client?

My SQL injection I was more thinking about a source of bugs, than a attack.

Maybe it's what you were asking for

**Comment by jorgerojas26** [comment]
> > May i ask why SQL Injection is a concern to you in this app when it is just a SQL client?
> 
> My SQL injection I was more thinking about a source of bugs, than a attack.
> 
> Maybe it's what you were asking for

I understand. I appreciate the feedback. For now, i'll merge this PR just to resolve the connection problem and open another PR addressing your feedback.

Thank you very much.

**Comment by ccoVeille** [comment]
Sure

**Pull request merged**